draft-ietf-ppm-dap.md

```diff
@@ -1323,26 +1323,57 @@ the input share is marked as invalid with a corresponding ReportShareError
 error.
 
 Before beginning the preparation step, Aggregators are required to perform the
-following generic checks.
+following checks:
 
 1. Check that the decrypted input share can be decoded. If not, the input share
    MUST be marked as invalid with the error `unrecognized_message`.
 
-1. Check if the report has already been aggregated with this aggregation
-   parameter. If this check fails, the input share MUST be marked as invalid
-   with the error `report_replayed`. This is the case if the report was used in
-   a previous aggregate request and is therefore a replay.
+1. Check if the report is too far into the future. Implementors can provide for
+   some small leeway, usually no more than a few minutes, to account for clock
+   skew. If a report is rejected for this reason, the Aggregator SHOULD mark the
+   input share as invalid with the error `report_too_early`.
+
+1. Check if the report's timestamp has passed its task's `task_expiration` time,
+   if so the Aggregator MAY mark the input share as invalid with the error
+   "task_expired".
+
+1. Check if the PlaintextInputShare contains unrecognized extensions. If so,
+   then the Aggregator MUST mark the input share as invalid with error
+   "unrecognized_message".
+
+1. Check if the ExtensionType of any two extensions in PlaintextInputShare is
+   the same. If so, then the Aggregator MUST mark the input share as invalid
+   with error "unrecognized_message".
+
+1. Check if the report may still be aggregated with the current aggregation
```

Comment:
nit (grammar): Most of the items in this list follow the template: "Check if <SOMETHING BAD HAPPNES>." Can we change this to something like "Check if the aggregation parameter violates the validity scope of the report."

Reply:
Yeah make sense :)